From 149c7e1d715dbcb1b1c8f555e7428d64d33f215a Mon Sep 17 00:00:00 2001
From: Jens Axboe <axboe@kernel.dk>
Date: Wed, 1 May 2024 14:02:59 -0600
Subject: [PATCH 1/3] test: skip test case if underlying filesystem doesn't
 support O_DIRECT

This is a bad test setup as you need O_DIRECT for full coverage, but
just skip tests if we fail opening a file with O_DIRECT.

Origin: upstream, commit:149c7e1d715dbcb1b1c8f555e7428d64d33f215a
Forwarded: not-needed
Signed-off-by: Jens Axboe <axboe@kernel.dk>
---
 test/defer-tw-timeout.c | 5 +++++
 test/fixed-buf-merge.c  | 4 ++++
 2 files changed, 9 insertions(+)

diff --git a/test/defer-tw-timeout.c b/test/defer-tw-timeout.c
index f754ec8..4313eae 100644
--- a/test/defer-tw-timeout.c
+++ b/test/defer-tw-timeout.c
@@ -99,6 +99,11 @@ static int test_file(struct io_uring *ring, char *__fname)
 
 	fd = open(fname, O_RDONLY | O_DIRECT);
 	if (fd < 0) {
+		if (errno == EINVAL) {
+			if (!__fname)
+				unlink(fname);
+			return T_EXIT_SKIP;
+		}
 		perror("open");
 		if (!__fname)
 			unlink(fname);
diff --git a/test/fixed-buf-merge.c b/test/fixed-buf-merge.c
index 98943b5..fbc94a4 100644
--- a/test/fixed-buf-merge.c
+++ b/test/fixed-buf-merge.c
@@ -36,6 +36,10 @@ int main(int argc, char *argv[])
 
 	fd = open(filename, O_RDONLY | O_DIRECT, 0644);
 	if (fd < 0) {
+		if (errno == EINVAL) {
+			unlink(filename);
+			return T_EXIT_SKIP;
+		}
 		perror("open");
 		goto err_unlink;
 	}
-- 
2.43.0

